roofit/roofitcore/src/RooAbsTestStatistic.cxx

```diff
@@ -108,46 +95,25 @@ RooAbsTestStatistic::RooAbsTestStatistic(const char *name, const char *title, Ro
   _rangeName(cfg.rangeName),
   _addCoefRangeName(cfg.addCoefRangeName),
   _splitRange(cfg.splitCutRange),
-  _simCount(1),
   _verbose(cfg.verbose),
-  _nGof(0),
-  _gofArray(0),
-  _nCPU(cfg.nCPU),
-  _mpfeArray(0),
-  _mpinterl(cfg.interleave),
-  _doOffset(kFALSE),
-  _evalCarry(0)
+  // Determine if RooAbsReal is a RooSimultaneous
+  _gofOpMode{(cfg.nCPU>1 || cfg.nCPU==-1) ? MPMaster : (dynamic_cast<RooSimultaneous*>(_func) ? SimMaster : Slave)},
+  _nEvents{data.numEntries()},
+  _nCPU(cfg.nCPU != -1 ? cfg.nCPU : 1),
+  _mpinterl(cfg.interleave)
 {
   // Register all parameters as servers
-  RooArgSet* params = real.getParameters(&data) ;
-  _paramSet.add(*params) ;
-  delete params ;
-
-  if (_nCPU>1 || _nCPU==-1) {
-
-    if (_nCPU==-1) {
-      _nCPU=1 ;
-    }
-
-    _gofOpMode = MPMaster ;
-
-  } else {
-
-    // Determine if RooAbsReal is a RooSimultaneous
-    Bool_t simMode = dynamic_cast<RooSimultaneous*>(&real)?kTRUE:kFALSE ;
-
-    if (simMode) {
-      _gofOpMode = SimMaster ;
-    } else {
-      _gofOpMode = Slave ;
-    }
+  _paramSet.add(*std::unique_ptr<RooArgSet>{real.getParameters(&data)});
+
+  if (cfg.rangeName.find(',') != std::string::npos) {
+    throw std::runtime_error(
```

Comment:
I would suggest to print the problem also on the error stream using coutE(Topic). Not every OS prints what's wrong when an exception is not caught.

Maybe `invalid_argument` is more appropriate here? Does this happen based on arguments that are passed to `fitTo` or similar?

Reply:
Yes, this is the case. I'll take `invalid_argument` then and also follow your other suggestion!